### PR TITLE
Upgrade spring library part 2 (connect #2896)

### PR DESCRIPTION
**Pull request by muloem**
#### The solution

* Upgrade the spring and spring-security library versions
* Make necessary changes related to migration of the two versions

## Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/com/gallatinsystems/user/app/gwt/client/PermissionConstants.java [hunk 1]
```diff
@@ -37,6 +37,6 @@
     public static final String EDIT_TOKENS = "EDIT TOKENS";
     public static final String VIEW_MESSAGES = "VIEW MESSAGES";
     public static final String EDIT_IMMUTABLITY = "EDIT IMMUTABILITY";
-    public static final String ADMIN = "ADMIN";
-    public static final String SUPER_ADMIN = "SUPER_ADMIN";
+    public static final String ROLE_ADMIN = "ROLE_ADMIN";
```
Comment (valllllll2000) [inline GAE/src/com/gallatinsystems/user/app/gwt/client/PermissionConstants.java]: Why one is called ROLE_ADMIN? and the other SUPER_ADMIN, should it be called ROLE_SUPER_ADMIN?

Comment (stellanl) [reply GAE/src/com/gallatinsystems/user/app/gwt/client/PermissionConstants.java]: No, strangely super-admin is not a role.

**Comment by muloem** [comment]
The ROLE_SUPER_ADMIN is not used anywhere so I removed it completely otherwise yes I would have had to rename as you say.

**Comment by stellanl** [comment]
I still see ROLE_SUPER_ADMIN in the code.

**Comment by muloem** [comment]
> I still see ROLE_SUPER_ADMIN in the code.
In UserRestService.java, for example.

@stellanl yes but thats `AppRole.ROLE_SUPER_ADMIN` and not `PermissionConstants.ROLE_SUPER_ADMIN`

**Comment by muloem** [comment]
@stellanl @valllllll2000 so after some digging around turns out that we may not even need that PermissionConstants class.  It is pretty much not used anywhere.  The field `UserDto.admin` which is affected by that code is never passed to the UI nor to the backend.  I will make a follow up PR to remove it since I dont want to include that as part of this.

**Pull request merged**